### DISCUSSION TITLE
How to obtain the actual flight trajectory of multiple cfs?

### DISCUSSION THREAD

**Discussion by sadlim** [USC-ACTLab/crazyswarm#739] [Q&A] [answered]
We are using `vicon &crazyswarm 1`.
Previously, the `Unique Marker Arrangements` mode was used, so the actual flight trajectory of the CF (i.e. the CSV file of rigid body motion) can be obtained in the vicon tracker.
At present, we have switched to `Single Marker` mode, and the vicon tracker cannot export the trajectory of a single marker.
**Is there any other way to obtain it?**

**Comment by whoenig** [comment, marked as answer]
You can record a rosbag, since at least the tf will contain the trajectories. There is also a write_csv mode in https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/hover_swarm.launch#L41.

**Reply by sadlim** [reply]
Thank you!
